roles/openshift_logging/tasks/generate_certs.yaml

```diff
@@ -146,20 +112,30 @@
   register: truststore_jks
   check_mode: no
 
-- name: create JKS generation pod
+- set_fact:
+    jks_container_uuid: {{'abcdefghijklmnopqrstuvwxyz0123456789'|random_word(10)}}
+  when: not elasticsearch_jks.stat.exists or not logging_es_jks.stat.exists or not system_admin_jks.stat.exists or not truststore_jks.stat.exists
+
+- name: create JKS generation container
   command: >
-    {{ openshift.common.client_binary }} --config={{ mktemp.stdout }}/admin.kubeconfig create -f {{mktemp.stdout}}/jks_pod.yaml -n {{openshift_logging_namespace}} -o name
-  register: podoutput
-  check_mode: no
+    docker run
+    -u 0
+    -e "PROJECT={{openshift_logging_namespace}}"
+    -e "CERT_DIR={{generated_certs_dir}}"
+    -v "{{generated_certs_dir}}:{{generated_certs_dir}}"
+    --name "jks_gen_{{jks_container_uuid}}"
+    --entrypoint="/bin/bash"
+    "{{openshift_logging_image_prefix}}logging-deployer:{{openshift_logging_image_version}}"
+    "{{generated_certs_dir}}/generate-jks.sh"
+  register: container_output
+  check_mode: no
+  become: yes
   when: not elasticsearch_jks.stat.exists or not logging_es_jks.stat.exists or not system_admin_jks.stat.exists or not truststore_jks.stat.exists
 
 - command: >
-    {{ openshift.common.client_binary }} --config={{ mktemp.stdout }}/admin.kubeconfig get {{podoutput.stdout}} -o jsonpath='{.status.phase}' -n {{openshift_logging_namespace}}
+    docker wait "jks_gen_{{jks_container_uuid}}"
   register: result
-  until: result.stdout.find("Succeeded") != -1
-  retries: 5
-  delay: 10
-  changed_when: no
+  changed_when: result.rc == "0"
```

Comment:
The script for metrics appeared to work as expected but had a bunch of logs in stderr which I could not seem to pin down an actual error.  Additionally, the return code was zero so I don't know if there is a good way to determine success of failure.